packages/api/cms-api/src/generator/utils/ts-morph-helper.ts

```diff
@@ -61,7 +61,10 @@ function findImportPath(importName: string, generatorOptions: { targetDirectory:
 
 export function findEnumName(propertyName: string, metadata: EntityMetadata<any>): string {
     const tsProp = morphTsProperty(propertyName, metadata);
-    return tsProp.getType().getText(tsProp);
+    return tsProp
+        .getType()
+        .getText(tsProp)
+        .replace(/ ?\| ?undefined$/, "");
```

Comment:
I tried also with the following:
```
            if (tsType.isUnion() && tsType.getUnionTypes().length == 2 && tsType.getUnionTypes()[0].getText() == "undefined") {
                // undefinded | type (or prop?: type) -> type
                tsType = tsType.getUnionTypes()[1];
            }
```
but failed to get that working, because for enums the union is `undefined | EnumValue1 | EnumValue2` etc. So I decided to work with strings

Reply:
@nsams I found another way that works with my current implementation, but theoretically there is a chance that `undefined` might be returned (which wasn't the case for me).

What do you think about this: `tsProp.getTypeNode()?.getText()`?

Reply:
@nsams I've tested `propertyName: Xxx | undefined` with `getTypeNode()` and an exception is thrown. So I think that it's better to go with your string implementation because I prefer to support both versions as they are both valid implementations.

Reply:
@nsams your version even works with `propertyName: undefined | Xxx` (`undefined` first).